Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -1445,87 +1445,184 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::sort)
 // 23.1.3.31 Array.prototype.splice ( start, deleteCount, ...items ), https://tc39.es/ecma262/#sec-array.prototype.splice
 JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::splice)
 {
+    // 1. Let O be ? ToObject(this value).
     auto this_object = TRY(vm.this_value().to_object(vm));
 
+    // 2. Let len be ? LengthOfArrayLike(O).
     auto initial_length = TRY(length_of_array_like(vm, this_object));
 
+    // 3. Let relativeStart be ? ToIntegerOrInfinity(start).
     auto relative_start = TRY(vm.argument(0).to_integer_or_infinity(vm));
 
+    // 4. If relativeStart = -∞, let actualStart be 0.
+    // FIXME: Spec does not ask for relative_start to be changed
     if (Value(relative_start).is_negative_infinity())
         relative_start = 0;
 
+    // FIXME: Set to 0 if relative_start is -infinity
     u64 actual_start;
 
+    // FIXME: Slightly different from spec logic
+    // 5. Else if relativeStart < 0, let actualStart be max(len + relativeStart, 0).
     if (relative_start < 0)
         actual_start = max((ssize_t)initial_length + relative_start, (ssize_t)0);
+    // 6. Else, let actualStart be min(relativeStart, len).
     else
         actual_start = min(relative_start, initial_length);
 
+    // 7. Let itemCount be the number of elements in items.
+    // REMARK: 0 if no items, updated later if there are items
```

Comment:
We call these NOTE, but this also seems fairly obvious from the name

Reply:
Good to know :)